gateway/src/main/kotlin/com/example/gateway/mapper/OrderProtoMapper.kt

```diff
@@ -0,0 +1,179 @@
+package com.example.gateway.mapper
+
+import com.example.delivery.dto.request.CreateOrderDTO
+import com.example.delivery.dto.request.CreateOrderItemDTO
+import com.example.delivery.dto.request.UpdateOrderDTO
+import com.example.delivery.dto.response.OrderDTO
+import com.example.delivery.dto.response.OrderItemDTO
+import com.example.delivery.dto.response.OrderItemWithProductDTO
+import com.example.delivery.dto.response.OrderWithProductDTO
+import com.example.delivery.dto.response.ShipmentDetailsDTO
+import com.example.delivery.exception.OrderNotFoundException
+import com.example.delivery.exception.ProductAmountException
+import com.example.delivery.exception.ProductNotFoundException
+import com.example.delivery.exception.UserNotFoundException
+import com.example.gateway.mapper.ProductProtoMapper.toDTO
+import com.example.internal.commonmodels.order.order.Order
+import com.example.internal.commonmodels.order.order_item.OrderItem
+import com.example.internal.commonmodels.order.shipment_details.ShipmentDetails
+import com.example.internal.input.reqreply.order.create.CreateOrderRequest
+import com.example.internal.input.reqreply.order.create.CreateOrderResponse
+import com.example.internal.input.reqreply.order.delete.DeleteOrderResponse
+import com.example.internal.input.reqreply.order.find.FindOrderByIdResponse
+import com.example.internal.input.reqreply.order.find_by_user_id.FindOrdersByUserIdResponse
+import com.example.internal.input.reqreply.order.update.UpdateOrderRequest
+import com.example.internal.input.reqreply.order.update.UpdateOrderResponse
+import com.example.internal.input.reqreply.order.update_status.UpdateOrderStatusResponse
+import java.math.BigDecimal
+
+object OrderProtoMapper {
+
+    @SuppressWarnings("ThrowsCount")
+    fun CreateOrderResponse.toDTO(): OrderDTO {
+        require(this != CreateOrderResponse.getDefaultInstance()) { "Acquired message is empty!" }
+        if (hasFailure()) {
+            when (failure.errorCase!!) {
+                CreateOrderResponse.Failure.ErrorCase.USER_NOT_FOUND ->
+                    throw UserNotFoundException(failure.message)
+
+                CreateOrderResponse.Failure.ErrorCase.PRODUCT_NOT_FOUND ->
+                    throw ProductNotFoundException(failure.message)
+
+                CreateOrderResponse.Failure.ErrorCase.PRODUCT_NOT_SUFFICIENT_AMOUNT ->
+                    throw ProductAmountException(
+                        failure.message
+                    )
+
+                CreateOrderResponse.Failure.ErrorCase.ERROR_NOT_SET ->
+                    throw IllegalStateException(failure.message)
```

Comment:
you can use function from **Preconditions.kt** to throw IllegalStateException
<img width="787" alt="image" src="https://github.com/user-attachments/assets/7f736f53-c243-48d3-bb11-a6aae6337f21">

```suggestion
                    error(failure.message)
```